client/Windows/wf_cliprdr.c

```diff
@@ -1348,9 +1354,19 @@ static void wf_cliprdr_array_ensure_capacity(wfClipboard* clipboard)
 {
 	if (clipboard->nFiles == clipboard->file_array_size)
 	{
-		clipboard->file_array_size *= 2;
-		clipboard->fileDescriptor = (FILEDESCRIPTORW**) realloc(clipboard->fileDescriptor, clipboard->file_array_size * sizeof(FILEDESCRIPTORW*));
-		clipboard->file_names = (WCHAR**) realloc(clipboard->file_names, clipboard->file_array_size * sizeof(WCHAR*));
+		int new_size;
+		FILEDESCRIPTORW **new_fd;
+		WCHAR **new_name;
+
+		new_size = clipboard->file_array_size * 2;
+		new_fd = (FILEDESCRIPTORW**) realloc(clipboard->fileDescriptor, new_size * sizeof(FILEDESCRIPTORW*));
+		new_name = (WCHAR**) realloc(clipboard->file_names, new_size * sizeof(WCHAR*));
+		if (!new_fd || !new_name)
+			return;
```

Comment:
if realloc(new_name) fails, we will have potentially an incorrect pointer in clipboard->fileDescriptor, and probably a leak of new_fd